libs/payments/ui/src/lib/actions/recordEmitterEvent.ts

```diff
@@ -41,11 +36,13 @@ async function recordEmitterEventAction(
     searchParams: flattenRouteParams(searchParams),
   };
 
-  return getApp().getActionsService().recordEmitterEvent({
-    eventName,
-    requestArgs,
-    paymentProvider,
-  });
+  return getApp()
+    .getActionsService()
+    .recordEmitterEvent({
+      eventName,
+      requestArgs,
+      paymentProvider: paymentProvider || undefined,
```

Comment:
`PaymentProvidersType` has no falsy subtype. Is this `|| undefined` cast necessary?